lib/chore/worker.rb

```diff
@@ -42,6 +42,24 @@ def expires_at
       @started_at + total_timeout
     end
 
+    def duplicate_work?(item)
+      # if we've got a duplicate, remove the message from the queue by not actually running and also not reporting any errors
+      payload = options[:payload_handler].payload(item.decoded_message)
+      
+      # only dedupe on the message id if we're getting work from an sqs consumer
+      return false if item.consumer.class != Chore::Queues::SQS::Consumer
+
+      # if the message is a duplicate on the queue id, re-enque it
```

Comment:
`re-enqueue`

Reply:
If we have a dupe on message id alone, why would we re-enqueue it? We'd want to silently kill this message right?

Reply:
No, no we don't. Right. If the message ids are the same, we just want to ignore this one because we're expecting a real job run to get through at some point.

We only want to ack if they have DIFFERENT message ids.